src/components/Forms/Input.jsx

```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+
+const Input = React.forwardRef(
+  (
+    {
+      type,
+      name,
+      disabled = false,
+      errors,
+      errorMessage,
+      children,
+      onChange,
+      ...props
+    },
+    ref
+  ) => {
+    const [active, setActive] = React.useState(false);
+
+    function handleActivation(e) {
+      setActive(!!e.target.value);
+      onChange(e);
+    }
+
+    return (
+      <>
+        <div
+          className={[
+            'relative border-2 rounded-md mb-2 text-background border-tertiary',
+            errors ? 'border-red-500' : '',
+          ].join(' ')}
+        >
+          <input
+            className={[
+              'outline-none w-full rounded bg-transparent text-sm transition-all duration-200 ease-in-out p-2',
+              active ? 'pt-6' : '',
+              errors ? 'error' : '',
+            ].join(' ')}
+            name={name}
+            type={type}
+            disabled={disabled}
+            onChange={handleActivation}
+            {...props}
+            ref={ref}
+          />
+
+          <label
+            className={[
+              'absolute top-0 left-0 flex items-center text-tertiary text-opacity-50 p-2 transition-all duration-200 ease-in-out',
+              active ? 'text-xs' : 'text-sm',
+              errors ? 'error' : '',
+            ].join(' ')}
+            htmlFor={name}
+          >
+            {children}
+          </label>
+        </div>
+        {errors && (
+          <span className="inline-flex text-sm error">{errors.message}</span>
+        )}
+      </>
+    );
+  }
+);
+
+export default Input;
```

Comment:
Interesting usage of a custom input, i liked it. One problem is that it does not get active when we click on the placeholder, can you fix that please ?

Reply:
it gets active when u start typing

Reply:
if you click on the placeholders nothing will happen, you should click on other part of the input to get active and focus

Reply:
+1 to what @ayubabdullah said.

Reply:
@sarahaboudalal add an id to the label, it will solve the issue